airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/MysqlJdbcEncryption.kt

```diff
@@ -0,0 +1,193 @@
+
+        if (!sslMode.isNullOrEmpty()) {
+            additionalParameters[SSL_MODE] = sslMode
+
+            caCertKeyStorePair = prepareCACertificateKeyStore()
+
+            if (null != caCertKeyStorePair) {
+                log.debug { "uri for ca cert keystore: ${caCertKeyStorePair.first}" }
+                try {
+                    additionalParameters.putAll(
+                        java.util.Map.of(
+                            TRUST_KEY_STORE_URL,
+                            caCertKeyStorePair.first.toURL().toString(),
+                            TRUST_KEY_STORE_PASS,
+                            caCertKeyStorePair.second,
+                            TRUST_KEY_STORE_TYPE,
+                            KEY_STORE_TYPE_PKCS12
+                        )
+                    )
+                } catch (e: MalformedURLException) {
+                    throw RuntimeException("Unable to get a URL for trust key store")
+                }
+            }
+
+            clientCertKeyStorePair = prepareClientCertificateKeyStore()
+
+            if (null != clientCertKeyStorePair) {
+                log.debug {
+                    "uri for client cert keystore: ${clientCertKeyStorePair.first} / ${clientCertKeyStorePair.second}"
+                }
+                try {
+                    additionalParameters.putAll(
+                        mapOf(
+                            CLIENT_KEY_STORE_URL to clientCertKeyStorePair.first.toURL().toString(),
+                            CLIENT_KEY_STORE_PASS to clientCertKeyStorePair.second,
+                            CLIENT_KEY_STORE_TYPE to KEY_STORE_TYPE_PKCS12
+                        )
+                    )
+                } catch (e: MalformedURLException) {
+                    throw RuntimeException("Unable to get a URL for client key store")
+                }
+            }
+        }
+
+        log.debug { "additional params: $additionalParameters" }
+        return additionalParameters
+    }
+
+    class Builder {
+        var sslMode: String = ""
+        var caCertificate: String = ""
+        var clientCertificate: String = ""
+        var clientKey: String = ""
+        var clientKeyPassword: String = ""
+
+        fun setSslMode(sslMode: String) = apply { this.sslMode = sslMode }
+        fun setCaCertificate(caCertificate: String) = apply { this.caCertificate = caCertificate }
+        fun setClientCertificate(clientCertificate: String) = apply {
+            this.clientCertificate = clientCertificate
+        }
+        fun setClientKey(clientKey: String) = apply { this.clientKey = clientKey }
+        fun setClientKeyPassword(clientKeyPassword: String) = apply {
+            this.clientKeyPassword = clientKeyPassword
+        }
+
+        fun build() =
+            MysqlJdbcEncryption(
+                sslMode,
+                caCertificate,
+                clientCertificate,
+                clientKey,
+                clientKeyPassword,
+            )
+    }
```

Comment:
This `Builder` class isn't necessary at all, you could have the constructor args of `MysqlJdbcEncryption` be `var`s instead. If that bothers you, rename that class to `MysqlJdbcSslPropertiesBuilder`.